osu.Game.Rulesets.Osu/Objects/Drawables/DrawableHitCircle.cs

```diff
@@ -88,50 +87,54 @@ protected override void UpdatePreemptState()
         {
             base.UpdatePreemptState();
 
-            ApproachCircle.FadeIn(Math.Min(TIME_FADEIN * 2, TIME_PREEMPT));
-            ApproachCircle.ScaleTo(1.1f, TIME_PREEMPT);
+            if (!Hidden)
+            {
+                ApproachCircle.FadeIn(Math.Min(TIME_FADEIN * 2, TIME_PREEMPT));
+                ApproachCircle.ScaleTo(1.1f, TIME_PREEMPT);
+            }
         }
 
         protected override void UpdateCurrentState(ArmedState state)
         {
-            double duration = ((HitObject as IHasEndTime)?.EndTime ?? HitObject.StartTime) - HitObject.StartTime;
-
-            glow.Delay(duration).FadeOut(400);
-
-            switch (state)
+            if (Hidden)
+                // TODO: figure out a smarter way to include the hit window so we are able to miss but don't keep
+                // it alive unnecessarily long.
+                this.FadeOut(TIME_PREEMPT * FadeOutSpeedMultiplier).Delay(ExpireAfter + 300).Expire();

```

Comment:
I really don't like this `+ 300`. The `.Delay()` is trying to keep it alive so we can click on it before it expires. But that only works up to the `HitObject.StartTime`. Clicking early would work but without the addition you wouldn't be able to miss circles (300ms = hit window). Any ideas to do it better?

Reply:
if anything it should be referencing the hit window value directly. i think it was previously, right?

Reply:
No, not in Lazer. Hit window values were referenced in stable snippets you gave me though. But at different locations.

Reply:
Ah you can access the hit windows. So I could adress this without any constants.